modules/jwplayerBidAdapter.js

```diff
@@ -50,4 +94,113 @@ export const spec = {
   // onBidderError: function({ error, bidderRequest }) {}
 };
 
+function buildRequest(bidRequest, bidderRequest) {
+  // Open RTB Request Object
+  const openrtbRequest = {
+    id: bidRequest.bidId,
+    imp: buildRequestImpression(bidRequest, bidderRequest),
+    site: buildRequestSite(bidRequest),
+    device: buildRequestDevice()
+  };
+
+  // Attaching GDPR Consent Params
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(openrtbRequest, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(openrtbRequest, 'regs.ext.gdpr', (bidderRequest.gdprConsent.gdprApplies ? 1 : 0));
+  }
+
+  // CCPA
+  if (bidderRequest.uspConsent) {
+    deepSetValue(openrtbRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+
+  return JSON.stringify(openrtbRequest);
+}
+
+function buildRequestImpression(bidRequest) {
+  const impressionObject = {
+    id: bidRequest.adUnitCode,
+  };
+
+  impressionObject.video = buildImpressionVideo(bidRequest);
+
+  const bidFloorData = buildBidFloorData(bidRequest);
+  if (bidFloorData) {
+    impressionObject.bidfloor = bidFloorData.floor;
+    impressionObject.bidfloorcur = bidFloorData.currency;
+  }
+
+  impressionObject.ext = buildImpressionExtension(bidRequest);
+
+  return [impressionObject];
+}
+
+function buildImpressionVideo(bidRequest) {
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  const video = {};
+
+  // Obtain all ORTB params related video from Ad Unit
+  VIDEO_ORTB_PARAMS.forEach((param) => {
+    if (videoParams.hasOwnProperty(param)) {
+      video[param] = videoParams[param];
+    }
+  });
+
+  return video;
+}
+
+function buildImpressionExtension(bidRequest) {
+  return {
+    appnexus: {
+      placement_id: bidRequest.params.placementId
+    }
+  };
+}
+
+function buildBidFloorData(bidRequest) {
+  const {params} = bidRequest;
+  const currency = params.currency || 'USD';
+
+  let floorData;
+  if (isFn(bidRequest.getFloor)) {
+    // Bid Floor
+    const bidFloorRequest = {
+      currency: currency,
+      mediaType: 'video',
```

Comment:
use the `VIDEO` const to reduce the string duplication. Remember, every char counts! Each letter is an extra byte in file size, and they add up quickly ;)